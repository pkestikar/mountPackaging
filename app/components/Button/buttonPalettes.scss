.mp--button-primary {
  background: $navy-blue;
  color: white;
  border: none;
  box-shadow: 0 2px 8px 0 $blue-grey;
}
.mp--button-primary:hover {
  box-shadow: 0 2px 8px 0 white;
}
.mp--button-primary:active {
  box-shadow: 0 2px 8px 0 white;
}

.mp--button-secondary {
  background: white;
  color: $navy-blue;
  border: 1px solid $navy-blue;
  box-shadow: 0 2px 8px 0 rgba(127, 133, 149, 0.28);
}
.mp--button-secondary:hover {
  box-shadow: 0 2px 8px 0 $navy-blue;
}
.mp--button-secondary:active {
  box-shadow: 0 2px 8px 0 $navy-blue;
}

.mp--button-tertiary {
  background: $dark-blue;
  color: white;
  border: 1px solid $dark-blue;
  box-shadow: 0 2px 8px 0 rgba(127, 133, 149, 0.28);
}
.mp--button-tertiary:hover {
  box-shadow: 0 2px 8px 0 $navy-blue;
}
.mp--button-tertiary:active {
  box-shadow: 0 2px 8px 0 $navy-blue;
}
