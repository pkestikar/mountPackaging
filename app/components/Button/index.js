import React from 'react';
import classnames from 'classnames';

const Button = (props) => {
    const componentName = 'mp--button';
    const classes = classnames(
        `${componentName}`,
        {[`${props.customClass}`]: !!props.customClass},
        {[`loading`]: !!props.isLoading},
        {['disabled']: !!props.disabled},
        {[`small`]: props.size === 'small'},
        {[`medium`]: props.size === 'medium'},
        {[`large`]: props.size === 'large'},
        {[`${componentName}-${props.palette}`]: !!props.palette}
    );

    return (
        <button className={classes} onClick={() => props.onClick()}>
            {props.isLoading ?
                'Loading..'
                :
                props.children
            }
        </button>
    )
};

Button.defaultProps = {
    size: 'medium',
    disabled: false,
    loading: false,
    palette: 'primary'
};

export default Button;
