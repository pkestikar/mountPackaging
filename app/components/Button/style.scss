@import './buttonPalettes';

.mp--button {
  height: 45px;
  width: 175px;
  font-family: $site-font-family;
  &.small {
    height: 35px;
    width: 145px;
  }
  &.medium {
    height: 45px;
    width: 175px;
  }
  &.large {
    height: 55px;
    width: 200px;
  }

  display: flex;
  justify-content: center;
  align-items: center;
  outline: none;
  transition: all 0.2s ease;

  &.disabled {
    //pointer-events: none;
    cursor: not-allowed;
  }
}

.mp--button:hover {
  cursor: pointer;
}
.mp--button:active {
  transform: scale(0.99);
}

