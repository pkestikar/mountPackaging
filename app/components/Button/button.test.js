import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';

import Button from './index';

describe('Testing out the Button component', () => {
    test('if the button renders correctly', () => {
        const wrapper = shallow(<Button>Button Text</Button>);

        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('should add in the appropriate color palette for the button based on the prop', () => {
        const wrapper = shallow(<Button palette={'secondary'}>Button Text</Button>);

        expect(toJSON(wrapper).props.className).toContain('mp--button-secondary');
    });

    it('should display Loading... when the isLoading prop is true', () => {
        const wrapper = shallow(<Button isLoading={true}>Button Text</Button>);


        expect(wrapper.find('Loading..').to.have.lengthOf(1));
    });
});
