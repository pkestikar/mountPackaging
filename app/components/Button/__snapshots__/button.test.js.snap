// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing out the Button component if the button renders correctly 1`] = `
<button
  className="mp--button medium mp--button-primary"
  onClick={[Function]}
>
  Button Text
</button>
`;
